test/index.js

```diff
@@ -0,0 +1,207 @@
+      const library = require('../src/index.js');
+      let subscription = {
+        endpoint: VALID_SUBSCRIPTION.endpoint,
+        keys: {
+          p256dh: VALID_SUBSCRIPTION.keys.p256dh
+        }
+      };
+
+      expect(
+        () => library.encrypt('Hello, World', subscription)
+      ).to.throw('Subscription is missing some encryption details');
+    });
+
+    it('should throw an error due to an invalid client public key', function() {
+      const library = require('../src/index.js');
+      expect(
+        () => library.encrypt('Hello, World', INVALID_P256DH_SUBSCRIPTION)
+      ).to.throw('Subscription\'s client key (p256dh) is invalid.');
+    });
+
+    it('should throw an error when no p256dh key is passed in', function() {
+      const library = require('../src/index.js');
+      let subscription = {
+        endpoint: VALID_SUBSCRIPTION.endpoint,
+        keys: {
+          auth: VALID_SUBSCRIPTION.keys.auth
+        }
+      };
+
+      expect(
+        () => library.encrypt('Hello, World', subscription)
+      ).to.throw('Subscription is missing some encryption details');
+    });
+
+    it('should return the correct encryption values', function() {
+      let crypto = require('crypto');
+
+      // This is for the salt
+      let saltStub = sinon.stub(crypto, 'randomBytes');
+      saltStub.withArgs(16).returns(new Buffer(EXAMPLE_SALT, 'base64'));
+      testStubs.push(saltStub);
+
+      // Server key generation
+      const exampleECDH = crypto.createECDH('prime256v1');
+      exampleECDH.generateKeys();
+      exampleECDH.setPrivateKey(EXAMPLE_SERVER_KEYS.private, 'base64');
+      exampleECDH.setPublicKey(EXAMPLE_SERVER_KEYS.public, 'base64');
+      // Make this a NOOP
+      exampleECDH.generateKeys = () => {
+        return exampleECDH.getPublicKey();
+      };
+      let ecdhStub = sinon.stub(crypto, 'createECDH');
+      ecdhStub.withArgs('prime256v1').returns(exampleECDH);
+      testStubs.push(ecdhStub);
+
+      const library = proxyquire('../src/index.js', {
+        'crypto': crypto
+      });
+
+      const response = library.encrypt(EXAMPLE_INPUT, VALID_SUBSCRIPTION);
+      Buffer.isBuffer(response.salt).should.equal(true);
+      response.salt.should.have.length(SALT_LENGTH);
+      response.salt.toString('base64').should.equal(EXAMPLE_SALT);
+
+      Buffer.isBuffer(response.serverPublicKey).should.equal(true);
+      response.serverPublicKey.should.have.length(SERVER_PUBLIC_KEY_LENGTH);
+      response.serverPublicKey.toString('base64').should.equal(EXAMPLE_SERVER_KEYS.public);
+
+      response.ciphertext.toString('base64').should.equal(EXAMPLE_OUTPUT);
+    });
+
+    it('should throw an error when the input is too large', function() {
+      const library = require('../src/index.js');
+
+      let largeInput = new Buffer(4081);
+      largeInput.fill(0);
+
+      expect(
+        () => library.encrypt(new Buffer(largeInput).toString('utf8'), VALID_SUBSCRIPTION)
```

Comment:
One last thing - `largeInput` is now a Buffer already, so no need to call `new Buffer()` here and in next `expect`